File: 110. Balanced Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool ans = true;

    int trav(TreeNode* node) {
        if (node == nullptr) return 0;

        int left = trav(node->left);
        int right = trav(node->right);

        if (abs(left - right) > 1) {
            ans = false;
        }

        return 1 + max(left, right);
    }

    bool isBalanced(TreeNode* root) {
        trav(root);
        return ans;
    }
};

File: 110. Balanced Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    boolean res = true;
    public int dfs(TreeNode node){
        if(node == null) return 0;
        
        int leftTree = dfs(node.left);
        int rightTree = dfs(node.right);

        if(Math.abs(leftTree - rightTree) > 1) res = false;
        
        return 1 + Math.max(leftTree, rightTree);
    }
    public boolean isBalanced(TreeNode root) {
        dfs(root);

        return res;
    }
}